engine/gameobject/src/gameobject/gameobject_script.cpp

```diff
@@ -797,6 +883,14 @@ namespace dmGameObject
      * go.set(url, "example.x", 7, {index=1})
      * ```
      *
+     * Set a material property array by a table of vector4
+     *
+     * ```lua
+     * -- set the first two vector4's in the array
```

Comment:
Is this setting the first two, or setting it to an array of two?
E.g. should we use the array index in "options" to let it know the start index?

Reply:
Currently, setting an array with a table will always start from the start of the array and set as many items as possible in the array. We could use the index to set a start element, but then perhaps we want the same for go.get for consistency which unfortunately is trickier. i.e the current API gets an element of the array by using go.get(..., "my_array", { index = 12 }) and so getting a sub-array probably requires some kind of range keys 🤔 

as in:
go.get(..., "my_array", { start = x, end = y  }) or { index = x, count = y } or something?